### ISSUE TITLE
service object: AutoGenerateThumbnail

### ISSUE THREAD

**Issue by elrayle** [projecthydra-labs/hydra-works#80]
Create a service in Hydra::Works at /lib/hydra/works/services/file/auto_generate_thumbnail.rb
```
module Hydra::Works
  class AutoGenerateThumbnail

    ##
    # Auto-generate a thumbnail for an existing file.
    #
    # @param [String] :path_to_source_file that serves as the base for the derivative thumbnail
    # @param [String] :path_to_target_directory where the generated thumbnail should be saved
    #
    # @return [String] the path with filename of the generated file

    def self.call( path_to_source_file, path_to_target_dir )

        # TODO write code to generate a thumbnail

  end
end
```

**Comment by awead** [comment]
Should the target param just be the GenericFile resource? You could also argue that thumbnail generation could be part of an upload service, when you're uploading your original content, but maybe you're trying to be more granular?

**Comment by awead** [comment]
Ok, https://gist.github.com/elrayle/8aa83c33cd31311f03c1 makes sense now.

**Comment by awead** [comment]
Use existing derivative services in Sufia - https://github.com/projecthydra/sufia/blob/7eaeef5e197b14b401e01fc52d55b1125a787f94/sufia-models/app/models/concerns/sufia/generic_file/derivatives.rb#L9-L21

**Comment by awead** [comment]
fixed in #105
